enterprise_access/apps/api_client/lms_client.py

```diff
@@ -195,3 +197,53 @@ def create_pending_enterprise_users(self, enterprise_customer_uuid, user_emails)
                 response.text,
             )
             raise exc
+
+    def get_pending_enterprise_group_memberships(self, enterprise_group_uuid):
+        """
+        Gets pending enterprise group memberships
+
+        Arguments:
+            enterprise_group_uuid (str): uuid of the enterprise group uuid
+
+        Returns:
+            A list of dicts of pecus in the form of that reminder emails should
+            be sent to:
+                {
+                    'pending_learner_id': integer,
+                    'enterprise_group_membership_uuid': UUID,
+                    'member_details': {
+                      'user_email': string,
+                    },
+                    'recent_action': string,
+                    'enterprise_customer': EnterpriseCustomerUser,
+                }
+        """
+        try:
+            url = f'{self.enterprise_group_membership}{enterprise_group_uuid}/learners/?pending_users_only=true'
+            results = []
+
+            while url:
```

Comment:
What's the purpose of the `while` statement?

Reply:
I think we need this `while` statement to check for the existence of a url in `next` from the json response since it is paginated. so let's say our response looked something like the `json` example below, we will reassign the url and loop through those results.
```

{
  "next": 'https://internal.courses.stage.edx.org/enterprise/api/v1/enterprise-group/{enterprise_group_uuid}/learners/?pending_users_only=true',
  "previous": null,
  "count": 200,
  "num_pages": 1,
  "current_page": 1,
  "start": 0,
  "results": [
    {
      "learner_id": 4813,
      "pending_learner_id": null,
      "enterprise_group_membership_uuid": null,
      "member_details": {
        "user_email": "user2@edx.org",
        "user_name": "User 2"
      },
      "recent_action": "Invited: April 04, 2024",
      "member_status": "accepted"
    },
  //....
  ]
}
```